Sources/DataPipeline/Config/SDKConfigBuilder.swift

```diff
@@ -137,12 +168,22 @@ public class SDKConfigBuilder {
         return self
     }
 
+    @available(iOSApplicationExtension, unavailable)
     public func build() -> SDKConfigBuilderResult {
-        // create `SdkConfig`` from given configurations
+        // create `SdkConfig` from given configurations
         let sdkConfig = SdkConfig.Factory.create(
             logLevel: logLevel
         )
 
+        // create plugins based on given configurations
+        var configuredPlugins: [Plugin] = []
+        if autoTrackUIKitScreenViews {
+            configuredPlugins.append(AutoTrackingScreenViews(
+                filterAutoScreenViewEvents: filterAutoScreenViewEvents,
+                autoScreenViewBody: autoScreenViewBody
+            ))
+        }
```

Comment:
`autoConfiguredPlugins` allows us providing more config options in builder to help customers setup the SDK. We don't need to pass and store all these configs in `DataPipelineConfigOptions` again. We can instead translate the config into a plugin, and `DataPipelineImplementation` will automatically attach all these plugins whenever initialized.

Open to suggestions though. And if this sounds good, I can do the changes for Logger plugin also to keep it consistent.